report_docx/report/report_docx.py

```diff
@@ -0,0 +1,306 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.report.report_sxw import report_sxw
+from openerp import pooler
+import logging
+from pyPdf import PdfFileWriter, PdfFileReader
+from reportlab.pdfgen import canvas
+import os
+import base64
+from docxtpl import DocxTemplate
+from openerp.tools.translate import _
+_logger = logging.getLogger(__name__)
+
+
+class ReportDocx(report_sxw):
+    def create(self, cr, uid, ids, data, context=None):
+        self.pool = pooler.get_pool(cr.dbname)
+        self.cr = cr
+        self.uid = uid
+        report_obj = self.pool.get('ir.actions.report.xml')
+        report_ids = report_obj.search(
+            cr, uid, [('report_name', '=', self.name[7:])], context=context)
+
+        if report_ids:
+            report_xml = report_obj.browse(
+                cr, uid, report_ids[0], context=context)
+            self.title = report_xml.name
+            if report_xml.report_type == 'docx':
+                return self.create_source_docx(cr, uid, ids, context)
+        return super(ReportDocx, self).create(cr, uid, ids, data, context)
+
+    def create_source_docx(self, cr, uid, ids, context=None):
+        data = self.generate_docx_data(cr, uid, ids, context)
+
+        tmp_folder_name = '/tmp/docx_to_pdf/'
```

Comment:
IMO we should instead be utilizing the python temp file lib instead of any hard coded paths.